### PR TITLE
mac_client docs

### PR DESCRIPTION
**Pull request by ashley-cui**
add doc explaining setup for podman on mac in current stage of dev

Signed-off-by: Ashley Cui <ashleycui16@gmail.com>

### CODE REVIEW

**Comment by ashley-cui** [comment]
@rhatdan @jwhonce PTAL

File: mac_client.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Podman Mac CLient tutorial
```
Comment (rhatdan) [inline mac_client.md]: CLient->Client

File: mac_client.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+First and foremost, the Mac Client is under heavy development. We are working on getting the
+Mac client to be packaged and run for a native-like experience. This is the setup tutorial
+for the Mac client at its current stage of development and packaging.
+
+The purpose of the Podman for Mac is to allow users to run Podman on a Mac. Since Podman is a Linux
+container runtime, The Mac client is actually a version of the [remote-client](remote_client.md),
+edited to that the client side works on a Mac machine, and connects to a Podman "backend" on a Linux
+machine, virtual or physical. The goal is to have a naitive-like experience when working with the Mac
+client, so the command line interface of the remote client is exactly thesame as the regular Podman
```
Comment (rhatdan) [inline mac_client.md]: thesame->the same

File: mac_client.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+commands with the exception of some flags being removed as they do not apply to the Mac client.
+
+## What you need
+To use the Mac client, you will need a binary built for MacOS and a Podman "backend" on a Linux machine;
+hereafter referred to as the Podman node. In this context, a Podman node is a Linux system with Podman
+installed on it and the varlink service activated.  You will also need to be able to ssh into this
+system as a user with privileges to the varlink socket (more on this later).
+
+## Getting the Mac client
+The Mac client is availible through [Homebrew](https://brew.sh/).
```
Comment (rhatdan) [inline mac_client.md]: available

File: mac_client.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+## Setting up the client and Podman node connection
+
+To use the Mac client, you must perform some setup on both the your Mac and Podman nodes. In this case,
+the Mac node refers to the Mac on which Podman is being run; and the Podman node refers to where
+Podman and its storage reside.
+
+### Connection settings
+Your Linux box must have ssh enabled, and you must copt your Mac's public key from `~/.ssh/id.pub` to
```
Comment (rhatdan) [inline mac_client.md]: copt?

Comment (rhatdan) [reply mac_client.md]: Mention ssh-copy-id

File: mac_client.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+`/root/.ssh/authorized_keys` on your Linux box. This allows for the use of SSH keys for remote access.
+
+You may need to edit your `/etc/ssh/sshd_config` in your Linux machine as follows:
+```
+PermitRootLogin yes
+```
+
+Use of SSH keys are strongly encouraged over regular login. However, if you wish toavoid ‘logging in’ every
```
Comment (rhatdan) [inline mac_client.md]: to avoid

File: mac_client.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+### Required permissions
+For now, the Mac client requires that you be able to run a privileged Podman and have privileged ssh
+access to the remote system.  This limitation is being worked on.
+
+
+#### Running the remote client
+There are three different ways to pass connection information into the client: flags, config file, and
+environment variables. All three require information on username and a remote host ip address. Most often,
+your username should be root and you can obtain your linux-ip using `ip addr`
```
Comment (rhatdan) [inline mac_client.md]: linux-ip -> remote host ip

File: mac_client.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+```
+If the conf file is set up, you may simply use podman as you wold on the linux machine. Take a look at
+[podman-remote.conf.5.md](https://github.com/containers/libpod/blob/master/docs/podman-remote.conf.5.md) on how to use the conf file:
+
+```
+$ podman images
+REPOSITORY                     TAG      IMAGE ID       CREATED         SIZE
+docker.io/library/ubuntu       latest   47b19964fb50   2 weeks ago     90.7 MB
+docker.io/library/alpine       latest   caf27325b298   3 weeks ago     5.8 MB
+quay.io/cevich/gcloud_centos   latest   641dad61989a   5 weeks ago     489 MB
```
Comment (rhatdan) [inline mac_client.md]: remove this name.  Perhaps add some redhat images.

File: mac_client.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
+# Podman Mac CLient tutorial
+
+## What is the Podman Mac Client
+
+First and foremost, the Mac Client is under heavy development. We are working on getting the
+Mac client to be packaged and run for a native-like experience. This is the setup tutorial
+for the Mac client at its current stage of development and packaging.
+
+The purpose of the Podman for Mac is to allow users to run Podman on a Mac. Since Podman is a Linux
+container runtime, The Mac client is actually a version of the [remote-client](remote_client.md),
```
Comment (rhatdan) [inline mac_client.md]: s/runtime/engine

Comment (rhatdan) [reply mac_client.md]: s/remote-client/podman-remote client

File: mac_client.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+edited to that the client side works on a Mac machine, and connects to a Podman "backend" on a Linux
+machine, virtual or physical. The goal is to have a naitive-like experience when working with the Mac
```
Comment (rhatdan) [inline mac_client.md]: s/naitive/native

File: mac_client.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+Your Linux box must have ssh enabled, and you must copt your Mac's public key from `~/.ssh/id.pub` to
+`/root/.ssh/authorized_keys` on your Linux box. This allows for the use of SSH keys for remote access.
+
+You may need to edit your `/etc/ssh/sshd_config` in your Linux machine as follows:
+```
+PermitRootLogin yes
+```
+
+Use of SSH keys are strongly encouraged over regular login. However, if you wish toavoid ‘logging in’ every
+time you run a podman command, you may edit your `/etc/ssh/sshd_config` on your Linux machine as follows:
```
Comment (rhatdan) [inline mac_client.md]: s/podman/Podman

File: mac_client.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+To connect using flags, you can use
+```
+$ podman --remote-host linux-ip --username root images
+REPOSITORY                     TAG      IMAGE ID       CREATED         SIZE
+docker.io/library/ubuntu       latest   47b19964fb50   2 weeks ago     90.7 MB
+docker.io/library/alpine       latest   caf27325b298   3 weeks ago     5.8 MB
+quay.io/cevich/gcloud_centos   latest   641dad61989a   5 weeks ago     489 MB
+k8s.gcr.io/pause               3.1      da86e6ba6ca1   14 months ago   747 kB
+```
+If the conf file is set up, you may simply use podman as you wold on the linux machine. Take a look at
```
Comment (rhatdan) [inline mac_client.md]: s/podman/Podman
s/wold/would

File: mac_client.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+Mac client to be packaged and run for a native-like experience. This is the setup tutorial
+for the Mac client at its current stage of development and packaging.
+
+The purpose of the Podman for Mac is to allow users to run Podman on a Mac. Since Podman is a Linux
+container runtime, The Mac client is actually a version of the [remote-client](remote_client.md),
+edited to that the client side works on a Mac machine, and connects to a Podman "backend" on a Linux
+machine, virtual or physical. The goal is to have a naitive-like experience when working with the Mac
+client, so the command line interface of the remote client is exactly thesame as the regular Podman
+commands with the exception of some flags being removed as they do not apply to the Mac client.
+
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: Which flags?  I'd add a parenthetical naming the flags or a pointer to where to find the information later in this doc.

File: mac_client.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+## What you need
+To use the Mac client, you will need a binary built for MacOS and a Podman "backend" on a Linux machine;
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: is there a minimum version of MacOS?  I think not, but thought I'd ask just in case.

File: mac_client.md (hunk 10 above)
Comment (TomSweeneyRedHat) [inline mac_client.md]: s/every time/time/

File: mac_client.md (hunk 5 above)
Comment (TomSweeneyRedHat) [reply mac_client.md]: suggest:  "Use of SSH keys are strongly encouraged to ensure a secure login.

File: mac_client.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+Use of SSH keys are strongly encouraged over regular login. However, if you wish toavoid ‘logging in’ every
+time you run a podman command, you may edit your `/etc/ssh/sshd_config` on your Linux machine as follows:
+```
+PasswordAuthentication no
+PermitRootLogin without-password
+```
+
+### Podman node setup
+The Podman node must have Podman (not the Mac client) installed as normal. If your system uses systemd,
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: suggest:
The Podman node must be running a Linux distribution that supports Podman and must have Podman (not the Mac client) installed.

Comment (TomSweeneyRedHat) [reply mac_client.md]: How would they know if their system uses systemd?

File: mac_client.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+## Getting the Mac client
+The Mac client is availible through [Homebrew](https://brew.sh/).
+```
+$ brew cask install podman
+```
+
+## Setting up the client and Podman node connection
+
+To use the Mac client, you must perform some setup on both the your Mac and Podman nodes. In this case,
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: s/the Mac/the your Mac/

File: mac_client.md (hunk 14 above)
Comment (TomSweeneyRedHat) [reply mac_client.md]: Probably worth squeezing in a note somewhere that you'll need root access for the setup stuff on the Podman node.

File: mac_client.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+command:
+```
+$ sudo podman --log-level debug varlink --timeout 0  unix://run/podman/io.podman
+```
+
+### Required permissions
+For now, the Mac client requires that you be able to run a privileged Podman and have privileged ssh
+access to the remote system.  This limitation is being worked on.
+
+
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: remove extra line

File: mac_client.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+#### Running the remote client
+There are three different ways to pass connection information into the client: flags, config file, and
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: you reference "config file" here and "conf file" below, I'd stick with "conf file" through out.

File: mac_client.md (hunk 7 above)
Comment (TomSweeneyRedHat) [reply mac_client.md]: quay.io/podman/stable  perhaps?

**Comment by TomSweeneyRedHat** [comment]
Great start @ashley-cui!  Most comments are of the nit variety.

**Comment by rhatdan** [comment]
Big documentation changes like this are usually a lot easier to write and comment on inside of google docs then they are in github.

**Comment by ashley-cui** [comment]
> Big documentation changes like this are usually a lot easier to write and comment on inside of google docs then they are in github.

Got it for next time :)

File: mac_client.md [hunk 18]
```diff
@@ -0,0 +1,99 @@
+# Podman Mac Client tutorial
+
+## What is the Podman Mac Client
+
+First and foremost, the Mac Client is under heavy development. We are working on getting the
+Mac client to be packaged and run for a native-like experience. This is the setup tutorial
+for the Mac client at its current stage of development and packaging.
+
+The purpose of the Podman for Mac is to allow users to run Podman on a Mac. Since Podman is a Linux
```
Comment (TomSweeneyRedHat) [inline mac_client.md]: s/the Mac Client for Podman/the Podman for Mac/

File: mac_client.md (hunk 6 above)
Comment (TomSweeneyRedHat) [reply mac_client.md]: maybe "obtain your Podman node's ip using `ip addr` on the node."

**Comment by TomSweeneyRedHat** [comment]
Other than the last linux-ip comment from @rhatdan, LGTM.

File: mac_client.md [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+## Getting the Mac client
+The Mac client is available through [Homebrew](https://brew.sh/).
+```
+$ brew cask install podman
+```
+
+## Setting up the client and Podman node connection
+
+To use the Mac client, you must perform some setup on both the the Mac and Podman nodes. In this case,
```
Comment (cfelder) [inline mac_client.md]: duplication

> ... the the Mac

**Comment by TomSweeneyRedHat** [comment]
LGTM, ship it when the CI behaves.

File: mac_client.md (hunk 12 above)
Comment (rhatdan) [reply mac_client.md]: Lets do this in  a future PR

**Comment by rhatdan** [comment]
/lgtm
/hold
Do a hold cancel when the tests complete.

**Comment by ashley-cui** [comment]
@rhatdan tests passed
/hold cancel

**Comment by ashley-cui** [comment]
/hold cancel

@rhatdan tests passed, need an appprove

**Pull request merged**